tap_klaviyo/utils.py

```diff
@@ -179,68 +179,113 @@ def authed_get(source, url, params):
-        for response in get_all_using_next(
-                stream['stream'], url, api_key,
-                latest_event_time):
+        params = {
+            "filter": f"equals(metric_id,\"{stream['tap_stream_id']}\"),greater-or-equal(timestamp,{latest_event_time})",
+            "include": "profile,metric",
+            "sort": "datetime"
+        }
+        for response in get_all_using_next(stream['stream'], endpoint, headers, params):
             events = response.json().get('data')
 
             if events:
+                included = response.json().get('included', [])
                 counter.increment(len(events))
-                transfrom_and_write_records(events, stream)
+                transfrom_and_write_records(events, stream, included, params.get("include","").split(","))
                 update_state(state, stream['stream'], get_latest_event_time(events))
                 singer.write_state(state)
 
     return state
 
-def get_full_pulls(resource, endpoint, api_key):
+def get_full_pulls(resource, endpoint, headers):
 
     with metrics.record_counter(resource['stream']) as counter:
-
-        for response in get_all_pages(resource['stream'], endpoint, api_key):
-            
-            records = response.json().get('data')
-            counter.increment(len(records))
-            transfrom_and_write_records(records, resource)
-
-
-def transfrom_and_write_records(events, stream):
+        if resource['stream'] == 'campaigns':
+            # params list campaigns stream
+            params_list = [
+                {
+                    "filter": "equals(messages.channel,'email')",
+                    "include": "tags,campaign-messages"
+                }
+            ]
+        elif resource['stream'] == 'global_exclusions':
+            # params list for global_exclusions stream
+            params_list = [
+                {
+                    "filter": "equals(subscriptions.email.marketing.suppression.reason,'HARD_BOUNCE')",
+                    "additional-fields[profile]": "subscriptions,predictive_analytics"
+                },
+                {
+                    "filter": "equals(subscriptions.email.marketing.suppression.reason,'USER_SUPPRESSED')",
+                    "additional-fields[profile]": "subscriptions,predictive_analytics"
+                },
+                {
+                    "filter": "equals(subscriptions.email.marketing.suppression.reason,'UNSUBSCRIBE')",
+                    "additional-fields[profile]": "subscriptions,predictive_analytics"
+                },
+                {
+                    "filter": "equals(subscriptions.email.marketing.suppression.reason,'INVALID_EMAIL')",
+                    "additional-fields[profile]": "subscriptions,predictive_analytics"
+                }
+            ]
+        else:
+            # params list for lists stream 
+            params_list = [
+                {
+                    "include": "tags"
+                }
+            ]
```

Comment:
## Optional
Move these params list to the class declaration or to a dict mapping of the stream so this block will not be required

Reply:
Done